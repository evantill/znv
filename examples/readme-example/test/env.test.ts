import {describe, expect, test} from '@jest/globals';
import {z} from "zod";
import {parseEnv} from "znv";

describe("testing using latest zod version", () => {

  test('readme example with a valid environment', () => {
    //given
    const env = {
      NICKNAME: 'evantill',
      LLAMA_COUNT: '666',
      COLOR: 'red'
    }
    //when
    const {NICKNAME, LLAMA_COUNT, COLOR, SHINY} = parseEnv(env, {
      NICKNAME: z.string().min(1),
      LLAMA_COUNT: z.number().int().positive(),
      COLOR: z.enum(["red", "blue"]),
      SHINY: z.boolean().default(true),
    });
    //then
    const actual = [NICKNAME, LLAMA_COUNT, COLOR, SHINY].join(", ");
    expect(actual).toBe('evantill, 666, red, true');
  });

  test('readme example with an empty environment', () => {
    //given
    const env = {}
    //when
    const parseInvalidEnv = () => {
      parseEnv(env, {
        NICKNAME: z.string().min(1),
        LLAMA_COUNT: z.number().int().positive(),
        COLOR: z.enum(["red", "blue"]),
        SHINY: z.boolean().default(true),
      });
    }
    //then
    expect(parseInvalidEnv).toThrowErrorMatchingSnapshot()
  })

});
