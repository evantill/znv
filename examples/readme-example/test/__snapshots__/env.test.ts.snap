// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing using latest zod version readme example with an empty environment 1`] = `
"[31mErrors found while parsing environment:[39m
  [[33mNICKNAME[39m]:
    This field is required.
    (received [36mundefined[39m)

  [[33mLLAMA_COUNT[39m]:
    This field is required.
    (received [36mundefined[39m)

  [[33mCOLOR[39m]:
    This field is required.
    (received [36mundefined[39m)
"
`;
